c/tskit/trees.c

```diff
@@ -2076,6 +2076,32 @@ typedef struct {
     const tsk_id_t *index_tuples;
 } indexed_weight_stat_params_t;
 
```

Comment:
I pulled this function out because it's repeated in `tsk_treeseq_two_locus_general_stat`